client/my-sites/site-settings/publishing-tools/post-by-voice.tsx

```diff
@@ -0,0 +1,108 @@
+import { Button } from '@automattic/components';
+import { localizeUrl } from '@automattic/i18n-utils';
+import { ToggleControl } from '@wordpress/components';
+import { useTranslate } from 'i18n-calypso';
+import { connect } from 'react-redux';
+import FormSettingExplanation from 'calypso/components/forms/form-setting-explanation';
+import SupportInfo from 'calypso/components/support-info';
+import { useSelector } from 'calypso/state';
+import { errorNotice, successNotice } from 'calypso/state/notices/actions';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+import { useGetPostByVoice } from './hooks/use-get-post-by-voice';
+import { useRegeneratePostByVoiceMutation } from './hooks/use-regenerate-post-by-voice-mutation';
+import { useSwitchPostByVoiceMutation } from './hooks/use-switch-post-by-voice-mutation';
+
+import './style.scss';
+
+type PostByVoiceSettingComponentProps = {
+	successNotice: typeof successNotice;
+	errorNotice: typeof errorNotice;
+};
+
+const PostByVoiceSettingComponent = ( {
+	successNotice,
+	errorNotice,
+}: PostByVoiceSettingComponentProps ) => {
+	const translate = useTranslate();
+	const selectedSiteId = useSelector( getSelectedSiteId );
+	const { data: postByVoiceSettings } = useGetPostByVoice( selectedSiteId );
+	const { mutate: switchPostByVoice, isPending: isPendingSwitch } =
+		useSwitchPostByVoiceMutation( selectedSiteId );
+	const { mutate: regeneratePostByVoice, isPending: isPendingRegenerate } =
+		useRegeneratePostByVoiceMutation( selectedSiteId );
+
+	const handleSwitch = ( checked: boolean ) => {
+		switchPostByVoice( checked, {
+			onSuccess: () => {
+				successNotice( translate( 'Changes saved successfully!' ) );
```

Comment:
```suggestion
				successNotice( translate( 'Settings saved successfully!' ), {
					duration: 6000,
				} )
```
The `6000` number is arbitrary, but it looks like the notices stick around indefinitely unless we pass a `duration`. This is such ephemeral information, so I think let's pass _some duration_.

We should do this for all the notices.